modules/scripts/genindex.py

```diff
@@ -93,19 +93,29 @@ def get_index(root, subprojects, sub_docpaths):
         toctree = get_toctree(os.curdir, [sub_index_path],
                               _title(subproject.split(os.path.sep)[0]))
         index.append(toctree)
+    # add javadoc if javadoc exist
+    if javadoc != "":
+        index = add_javadoc(index)
 
     return '\n\n'.join(index) + '\n'
 
+def add_javadoc(index):
+    javadoc = ["\n", "API Documentation", "\n", "=================", "\n",
```

Comment:
The first and last `\n` can be removed as we are already joining it via `\n\n` and appending `\n` at end